Rubberduck.Core/Templates/Template.cs

```diff
@@ -16,19 +18,37 @@ public Template(string name, ITemplateFileHandler handler)
         {
             _handler = handler;
 
-            Name = name;
-            IsUserDefined = VerifyIfUserDefined(name);
+            Name = name.EndsWith(".rdt") ? name.Substring(0, name.Length - 4) : name;
+            IsUserDefined = VerifyIfUserDefined(Name);
```

Comment:
This is comment will likely betray my lack of familiarity with using templates:  I could not find anything mandating an ".rdt" extension since the file content is a simple text file without any formatting expectations beyond the header text content itself (true?).  Is there a downside to accepting other extensions so long as the content is valid?

Reply:
Good question. Right now, I made up the extension `.rdt` just as a way to collect all valid files. Think of people making a `My Template.rdt.bak` copy of the `My Template.rdt`; the former should be ignored while the latter is listed. But I'm not set on this convention. If we want to allow any file, with any extension, to be listed, we can do that. It boils down to what we want to be listed or ignored, really.  I'm open to suggestions here.

Reply:
FYI - for the uninformed (like myself), searching for the meaning of *.rdt  points us at apps like `RemoteDesktop` and `Adobe Captivate`. :neutral_face: 

I'm guessing you intended 'RubberDuckTemplate' and  *.rdt is as good as any (IMO).  For those who know how to work with template files, I suspect the extension will not make them wonder if the files are anything more than simple text.

Reply:
While it's true that it's a plain text right now, there's no guarantee that it will remains so in future, esp. if it incorporates a macro system. This will depend on how we ultimately design the template and how it can be used. When it's a module, that's easy - just import it. But if it's a procedure or even a code fragment, then it can't be just a import routine.

Reply:
Does it have to be a 8.3 filename? If not, we can't go wrong with `.template` 😄

Reply:
There is no reason it should be restricted to 8.3 filename. I don't think our predeclared template even is equal or less than 8 characters. If nobody else has counterarguments or counteroffers, I can change to use `.template`.

Reply:
Actually, there may be an issue with changing the extension. We've already shipped `PredeclaredClassModule` template and therefore installations will already have a `PredeclaredClassModule.rdt` in their `Templates` folder. If we change the extension now, we may end up with both `PredeclaredClassModule.rdt` and `PredeclaredClassModule.template`. While I could use `template` only for user-defined template, it does make for inconsistent use of the extensions. Would you rather change the extension now and deal with duplicated `PredeclaredClassModule` or leave things alone?

Reply:
I'd rather change now, considering that 8.3 adherence is nothing that's really necessary and that `.template` is such a vast improvement over `.rdt`